### DISCUSSION TITLE
Misunderstanding in Computed Properties

### DISCUSSION THREAD

**Discussion by RealMrHex** [livewire/livewire#8500] [1. Help] [answered]
Based on the official documentation, computed properties allow you to access values and cache them for future access during a request:

> "Computed properties allow you to access values and cache them for future access during the request."

However, when I implemented the following computed property:

```php
#[Computed]
public function profile()
{
    return auth()->user()->profile;
}
```

I expected that calling the `profile()` function multiple times within a single request would result in only a single query being executed to fetch the profile. Subsequent calls should use the cached value.

For example:

```php
public function mount()
{
    $this->profile(); // should execute a query
    $this->profile(); // should use cache
    $this->profile(); // should use cache
}
```

However, when I check with Laravel Debugbar, I see that three queries are being executed instead of just one.

Here's what I've tried so far:

1. Verified the computed property syntax.
2. Added `persist: true` and a cache key to ensure caching is enabled.

Despite these attempts, the caching does not seem to work as expected. Is there something I am missing or misinterpreting about how computed properties are supposed to work in this context?

Any insights or suggestions would be greatly appreciated.

Thank you!

**Comment by danie-ramdhani** [comment, marked as answer]
You need to use that as properties:

```php
public function mount()
{
    $this->profile;
    $this->profile;
}
```

And you can see the difference using Laravel benchmark between using computed properties and call it as a function.

**Comment by RealMrHex** [comment]
Ugh, what a mistake!

Thank you so much @danie-ramdhani ❤️

**Reply by WanderingDaniel** [reply]
Well, your mistake helped me a year later, so thanks for that. 

Did the exact same thing :D
